framework/src/main/java/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1590,16 +1590,26 @@ private Object getValueOfExpressionInAnnotation(
         } else if (expr instanceof LongLiteralExpr) {
             return convert(((LongLiteralExpr) expr).asLong(), valueKind);
         } else if (expr instanceof UnaryExpr) {
-            if (((UnaryExpr) expr).getOperator() == UnaryExpr.Operator.MINUS) {
-                Object value =
-                        getValueOfExpressionInAnnotation(
-                                name, ((UnaryExpr) expr).getExpression(), valueKind);
-                if (value instanceof Number) {
-                    return convert((Number) value, valueKind, true);
-                }
+            switch (expr.toString()) {
+                    // Special-case the minimum values.  Separately parsing a "-" and a value
+                    // doesn't correctly handle the minimum values, because the absolute value of
+                    // the smallest member of an integral type is larger than the largest value.
+                case "-9223372036854775808L":
```

Comment:
Programmers can write the "L" in lowercase (even though they should not, since "l" looks like "1").